test/new-e2e/pkg/utils/secrets/client.go

```diff
@@ -0,0 +1,44 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
This looks quite specific to test your team will do (or limited number of teams would use. I think it would be better if you move this secret client to a folder in `tests/agent-subcommands` or `tests/agent-shared-component`. It will not prevent other teams from reusing this, but this is easier for us if you keep the ownership of this client.

Reply:
I've moved everything under `tests/agent-shared-components`